crates/bevy_ecs/src/world/deferred_world.rs

```diff
@@ -20,6 +20,8 @@ use super::{unsafe_world_cell::UnsafeWorldCell, Mut, World, ON_INSERT, ON_REPLAC
 
 /// A [`World`] reference that disallows structural ECS changes.
 /// This includes initializing resources, registering components or spawning entities.
+///
+/// This means that in order to add entities, for example, you will need to use commands instead of the world directly.
 pub struct DeferredWorld<'w> {
```

Comment:
I would also love to add a comment about why DeferredWorld exists and why it's sometimes a bad idea to, for example, spawn an entity directly from the world. If anyone knows please respond to this comment!

Reply:
My understanding is that there are some cases where it's not sound to hand out a `&mut World`. Instead, we hand out this struct, which has the same sort of API, and then *later* apply all of the things we have planned to it.